tests/bases/test_composition.py

```diff
@@ -557,3 +553,19 @@ def test_compositional_metrics_update():
 
     assert compos.metric_a._num_updates == 3
     assert compos.metric_b._num_updates == 3
+
+
+def test_compositional_metrics_forward():
+
+    compos = DummyMetric(5) + DummyMetric(4)
+
+    assert isinstance(compos, CompositionalMetric)
+    compos()
+    compos()
+    compos()
```

Comment:
lets also check that the return value is correct
```suggestion
    assert compos() == 9
    assert compos() == 9
    assert compos() == 9
```